examples/nvram/store.c

```diff
@@ -120,20 +147,49 @@ int TPM2_NVRAM_Store_Example(void* userCtx, int argc, char *argv[])
         nvAttributes, TPM2_DEMO_NV_TEST_SIZE, (byte*)gNvAuth, sizeof(gNvAuth)-1);
     if (rc != 0 && rc != TPM_RC_NV_DEFINED) goto exit;
 
-    printf("Storing key at TPM NV index 0x%x with password protection\n",
+    printf("Storing key at TPM NV index 0x%x with password protection\n\n",
              TPM2_DEMO_NVRAM_STORE_INDEX);
 
-    printf("Public part = %d bytes\n", keyBlob.pub.size);
-    rc = wolfTPM2_NVWriteAuth(&dev, &nv, TPM2_DEMO_NVRAM_STORE_INDEX,
-        (byte*)&keyBlob.pub.publicArea, keyBlob.pub.size, 0);
-    if (rc != 0) goto exit;
+    if (partialStore != PRIVATE_PART_ONLY) {
+        printf("Public part = %hu bytes\n", keyBlob.pub.size);
+        rc = wolfTPM2_NVWriteAuth(&dev, &nv, TPM2_DEMO_NVRAM_STORE_INDEX,
+            (byte*)&keyBlob.pub.size, sizeof(keyBlob.pub.size), 0);
+        if (rc != 0) goto exit;
+        printf("Stored 2-byte size marker before the private part\n");
+        offset += sizeof(keyBlob.pub.size);
+
+        /* Prepare temporary buffer */
+        packet.buf = pubAreaBuffer;
+        packet.pos = 0;
+        packet.size = sizeof(pubAreaBuffer);
+        /* Necessary for storing the publicArea with the correct byte aligness */
+        TPM2_Packet_AppendPublic(&packet, &keyBlob.pub);
```

Comment:
You will need to make this API public `WOLFTPM_API` to use from the example. I would instead recommend adding a new public API to handle converting a `TPM2B_PUBLIC` to byte stream. That API would have the `TPM2_Packet packet;` logic.

Reply:
I understand. Will fix.